src/fileLoader.ts

```diff
@@ -0,0 +1,446 @@
+    source.setAttribute('src', url)
+    source.setAttribute('type', fileType)
+    video.append(source)
+
+    /**
+     * Redactor will automatically insert a video element in a p tag,
+     * so we'll do it ourselves so that we can control its attributes.
+     */
+    return this.wrapElementInTag({
+      fsid,
+      fsName,
+      element: video,
+      tagName: 'p'
+    })
+  }
+
+  private createDownloadElement({ url, fsid, fsName }: CreateDownloadElementParams) {
+    const link = document.createElement('a')
+    link.setAttribute('fsid', fsid)
+    link.setAttribute('fsName', fsName)
+    link.setAttribute('ghost', 'true')
+    link.setAttribute('fscollapsable', 'true')
+    link.setAttribute('href', url)
+    link.textContent = `${fsName}`
+    return link
+  }
+
+  private createAudioElement({ url, fsid, fsName }: CreateAudioElementParams) {
+    const audio = document.createElement('audio')
+    audio.setAttribute('src', url)
+    audio.setAttribute('controls', 'true')
+    audio.setAttribute('fsid', fsid)
+    audio.setAttribute('fsName', fsName)
+    audio.setAttribute('fscollapsable', 'true')
+
+    return this.wrapElementInTag({
+      fsid,
+      fsName,
+      element: audio,
+      tagName: 'p'
+    })
+  }
+
+  private setStatus({ status, fsElement, fsid, removable = false }: SetStatusParams): Element | void {
+    if (fsid) {
+      const existingStatusElement = this.statusElementMapping[fsid]
+
+      if (existingStatusElement) {
+        existingStatusElement.remove()
+        delete this.statusElementMapping[fsid]
+      }
+    }
+
+    if (status) {
+      let element = document.createElement('label')
+
+      element.setAttribute('id', fsid)
+      element.setAttribute('ghost', 'true')
+      element.setAttribute('contenteditable', 'false')
+      element.style.fontWeight = 'bold'
+      element.textContent = status
+
+      if (removable) {
+        element.style.userSelect = 'all'
+      }
+
+      element = this.insertElementNearElement(element, fsElement) as HTMLLabelElement
+
+      if (fsid) {
+        this.statusElementMapping[fsid] = element
+      }
+
+      return element
+    }
+  }
+
+  public insertStatusAtCursor(status: string): string {
+    const identifier = Math.random().toString(36).substring(7)
+    this.setStatus({
+      status,
+      fsid: identifier
+    })
+    return identifier
+  }
+
+  public removeCursorStatus(identifier: string): void {
+    /**
+     * We want to search for the element based on identifier, because the actual element
+     * inserted may have been done so as raw HTML, and not via an element pointer.
+     */
+    const elements = this.options.getElementsBySelector(`#${identifier}`)
+
+    if (elements.length > 0) {
+      elements[0].remove()
+    }
+  }
+
+  public insertElementNearElement(domNodeToInsert: Element, inVicinityOfElement?: Element): Element | undefined {
+    if (!inVicinityOfElement) {
+      return
+    }
```

Comment:
What if `inVicinityOfElement` is `undefined`? Near which element are we going to insert `domNodeToInsert`?

Reply:
See line 438. Not sure. In JS code there is no return statement like there is here.

Reply:
See: https://github.com/standardnotes/editor-kit/blob/588ee2d98d034339c7704162acc0d0d961170438/src/lib/FileLoader.js#L257

It's calling `setStatus` with `fsElement` parameter as null. It will then call https://github.com/standardnotes/editor-kit/blob/588ee2d98d034339c7704162acc0d0d961170438/src/lib/FileLoader.js#L246 with `fsElement` as null. 

This line https://github.com/standardnotes/editor-kit/blob/588ee2d98d034339c7704162acc0d0d961170438/src/lib/FileLoader.js#L277 should be throwing an error.

Reply:
It's possible I made a mistake. I'd say just review this and make the call yourself because at this point you probably have more context than me :)

Reply:
Got it 😄